pkg/sql/sem/tree/datum.go

```diff
@@ -3844,54 +3847,82 @@ func (d *DEnum) Compare(ctx *EvalContext, other Datum) int {
 
 // Prev implements the Datum interface.
 func (d *DEnum) Prev(ctx *EvalContext) (Datum, bool) {
-	idx := d.EnumTyp.EnumGetIdxOfPhysical(d.PhysicalRep)
+	idx, err := d.EnumTyp.EnumGetIdxOfPhysical(d.PhysicalRep)
+	if err != nil {
+		panic(err)
+	}
 	if idx == 0 {
 		return nil, false
 	}
-	return MakeDEnumFromPhysicalRepresentation(
-		d.EnumTyp,
-		d.EnumTyp.TypeMeta.EnumData.PhysicalRepresentations[idx-1],
-	), true
+	enumData := d.EnumTyp.TypeMeta.EnumData
+	return &DEnum{
+		EnumTyp:     d.EnumTyp,
+		PhysicalRep: enumData.PhysicalRepresentations[idx-1],
+		LogicalRep:  enumData.LogicalRepresentations[idx-1],
+	}, true
 }
 
 // Next implements the Datum interface.
 func (d *DEnum) Next(ctx *EvalContext) (Datum, bool) {
-	idx := d.EnumTyp.EnumGetIdxOfPhysical(d.PhysicalRep)
-	physReps := d.EnumTyp.TypeMeta.EnumData.PhysicalRepresentations
-	if idx == len(physReps)-1 {
+	idx, err := d.EnumTyp.EnumGetIdxOfPhysical(d.PhysicalRep)
+	if err != nil {
+		panic(err)
+	}
+	enumData := d.EnumTyp.TypeMeta.EnumData
+	if idx == len(enumData.PhysicalRepresentations)-1 {
```

Comment:
nit, maybe...: this should be `idx + 1 >= len(enumData.PhysicalRepresentations)` for safety i reckon, in case we get idx == 0 and len(enumData.PhysicalRepresentations) == 0 (may be impossible but you never know).

Reply:
actuallly, meh....